### ISSUE TITLE
Normalize to Bibtex name format

### ISSUE THREAD

**Issue by stefan-kolb** [JabRef/jabref#318]
Rightclick formatter in authors field goes wrong.

Minimal example:
```
Stefan Kolb and J{\"o}rg Lenhard and Guido Wirtz
```
turns out to be
```
Kolb, Stefan and J{\"o}rg Lenhard and Wirtz, Guido
````

but should be

``` 
Kolb, Stefan and Lenhard, J{\"o}rg and Wirtz, Guido
```

- [x] `NameListNormalizer` should be transformed to a Formatter on that occasion.

On another note: is this really THE BibTex name format? This should help with multiple firstnames?!

**Comment by oscargus** [comment]
Yes, the Lastname, Firstname is THE BibTeX name format (if you should pick one). This is the only one that can handle Jr. and it also doesn't rely on "von" being spelled correctly (without capitals) for correct results.

A (possibly) related problem: The first name abbreviation for author column (and probably elsewhere) doesn't work as expected. Not sure if this happens to J{\"o}rg, but for sure H{\aa}kan is abbreviated to H{\aa}. instead of H.

**Comment by stefan-kolb** [comment]
What about Andreas Jr. {von Lastname}?

**Comment by oscargus** [comment]
Assuming his name is Andreas von Lastname, Jr. the correct way would be "von Lastname, Jr., Andreas". Your entry results in: von Lastname, A. J. or A. J. von Lastname.

"In BibTeX's world view, a name has four components:
1) First name (which includes any middle names provided)
2) von ("de la" or "van der" like components)
3) Last (Surname without the von part)
4) Jr (Things like "Jr.", "III", etc)"

http://nwalsh.com/tex/texhelp/bibtx-23.html says

 To summarize, you may type a name in one of three forms:
"First von Last" "von Last, First" "von Last, Jr, First"
You may almost always use the first form; you shouldn't if either there's a Jr part or the Last part has multiple tokens but there's no von part.

**Comment by oscargus** [comment]
There are more issues here:

```
Heng-Yu Jian and Xu, Z. and Chang, M.-C.F.
```
becomes
```
Heng-Yu Jian and Xu, Z. and Chang, M. -. C. F.
```
but I would expect
```
Jian, Heng-Yu and Xu, Z. and Chang, M.-C. F.
```

```
Oscar Gustafsson and Linda S. DeBrunner and Victor DeBrunner and H{\aa}kan Johansson
```
becomes
```
Gustafsson, Oscar and Linda S. DeBrunner and Victor DeBrunner and H{\aa}kan Johansson
```
while I would expect
```
Gustafsson, Oscar and DeBrunner,  Linda S. and DeBrunner, Victor and Johansson, H{\aa}kan
```

So, some observations/guesses:
 * Adding spaces after dots is good, unless there's a dash
 * Last names with double capital letters doesn't work (Victor Debrunner resulted in Debrunner, Victor, but Linda S. Debrunner remained the same)
 * First names with dashes doesn't work

**Comment by oscargus** [comment]
It is worth noting that the function producing the author string for the table column works better.

**Comment by koppor** [comment]
See https://github.com/JabRef/jabref/pull/319#issuecomment-155918893 - maybe test cases for `net.sf.jabref.model.entry.AuthorList` should be added testing for your input strings and whether they are parsed correctly.

**Comment by oscargus** [comment]
I think we should use BibtexNameFormatter, since that method is quite likely to get it correct. :-)
